### PR TITLE
fix(avatars): rectify `StatusIndicator` sizing and position

### PR DESCRIPTION
**Pull request by ze-flo**
## Description
Fix `StatusIndicator` styling to match [v7](https://606fa12ea4f2996a9cfd97aa--zendeskgarden.netlify.app/avatars/#text)

## Detail
- Allows `StatusIndicator` to stretch up to a `max-size`
- Fix `StatusIndicator` positioning for large `Avatar`

Screenshot 2024-08-20 at 7 59 01 PM
 
Screenshot 2024-08-20 at 7 58 46 PM
Screenshot 2024-08-20 at 7 59 36 PM
Screenshot 2024-08-20 at 7 59 23 PM

## Checklist

- [ ] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`npm start`)
- [ ] ~~:arrow_left: renders as expected with reversed (RTL) direction~~
- [ ] ~~:black_circle: renders as expected in dark mode~~
- [ ] ~~:metal: renders as expected with Bedrock CSS (`?bedrock`)~~
- [x] :guardsman: includes new unit tests. Maintain existing coverage (always >= 96%)
- [ ] ~~:wheelchair: tested for WCAG 2.1 AA accessibility compliance~~
- [ ] ~~:memo: tested in Chrome, Firefox, Safari, and Edge~~

### CODE REVIEW

File: packages/avatars/src/styled/StyledStatusIndicatorBase.ts [hunk 1]
```diff
@@ -41,7 +41,6 @@ const sizeStyles = (props: IStyledStatusIndicatorProps) => {
   return css`
     border: ${offset} ${props.theme.borderStyles.solid};
     border-radius: ${size};
-    width: ${size};
```
Comment (ze-flo) [inline packages/avatars/src/styled/StyledStatusIndicatorBase.ts:44]: Matches with [v7](https://github.com/zendeskgarden/css-components/blob/%40zendeskgarden/css-avatars%406.0.8/packages/avatars/src/_badge.css) CSS.

**Review by jzempel — DISMISSED** [review]
I think we can see from the Menu [patterns] story that I forgot to deal with the status indicator (`figcaption`) `box-shdow` in #1885. Also the #1897 update seems to yield unpredictable results if variable `surfaceColor` (ex "background.raised") is applied to the `Avatar` story before any other controls are set. Can we nab those here?

Screenshot 2024-08-21 at 6 07 11 PM

Screenshot 2024-08-21 at 6 16 50 PM

**Comment by geotrev** [comment]
> I think we can see from the Menu [patterns] story that I forgot to deal with the status indicator (`figcaption`) `box-shdow` in #1885. Also the #1897 update seems to yield unpredictable results if variable `surfaceColor` (ex "background.raised") is applied to the `Avatar` story before any other controls are set. Can we nab those here?
> 
> Screenshot 2024-08-21 at 6 07 11 PM Screenshot 2024-08-21 at 6 16 50 PM

The gift that keeps on giving

File: packages/avatars/src/styled/StyledAvatar.ts [hunk 2]
```diff
@@ -80,11 +80,12 @@ const colorStyles = ({
   }
 
   if ($status) {
-    surfaceColor = $surfaceColor?.includes('.')
-      ? getColor({ variable: $surfaceColor, theme })
-      : $surfaceColor || getColor({ variable: 'background.default', theme });
+    surfaceColor =
+      $surfaceColor && /^\w+\.\w+$/u.test($surfaceColor)
```
Comment (jzempel) [inline packages/avatars/src/styled/StyledAvatar.ts:84]: Is there a reason we moved away from the simpler `includes('.')` conditional here?

Comment (ze-flo) [reply packages/avatars/src/styled/StyledAvatar.ts:84]: Ooops, I forgot to clean up after myself. 🤦 This is what I wanted to bring up yesterday.

Will any of the [components](https://github.com/zendeskgarden/react-components/pull/1885) accepting a variable key as a user-defined color ever need to support `rgba` values created by using `getColor` with a `transparency` option?

Comment (jzempel) [reply packages/avatars/src/styled/StyledAvatar.ts]: Ah dang, that's the sneaky notation I hadn't thought of. Let's defer to address more holistically. (I recall @geotrev suggesting some kind if `isColorVariable` utility added to `theming`).

**Review by jzempel — APPROVED** [review]
Thanks for nabbing those finicky bits and adding tests so we don't keep regressing.

**Pull request merged**